pkg/sdn/plugin/proxy.go

```diff
@@ -87,29 +93,73 @@ func (proxy *OsdnProxy) watchEgressNetworkPolicies() {
 	})
 }
 
-func (proxy *OsdnProxy) updateNetworkPolicy(policy osapi.EgressNetworkPolicy) {
-	firewall := make([]proxyFirewallItem, len(policy.Spec.Egress))
+func (proxy *OsdnProxy) updateEgressNetworkPolicy(policy osapi.EgressNetworkPolicy) {
+	ns := policy.Namespace
+	firewall := make([]firewallItem, len(policy.Spec.Egress))
 	for i, rule := range policy.Spec.Egress {
 		_, cidr, err := net.ParseCIDR(rule.To.CIDRSelector)
 		if err != nil {
 			// should have been caught by validation
-			glog.Errorf("Illegal CIDR value %q in EgressNetworkPolicy rule", rule.To.CIDRSelector)
+			glog.Errorf("Illegal CIDR value %q in EgressNetworkPolicy rule for policy: %v", rule.To.CIDRSelector, policy.UID)
 			return
 		}
-		firewall[i] = proxyFirewallItem{rule.Type, cidr}
+		firewall[i] = firewallItem{rule.Type, cidr}
 	}
 
+	// Add/Update/Delete firewall rules for the namespace
 	if len(firewall) > 0 {
-		proxy.firewall[policy.Namespace] = firewall
+		updated := false
+		if ref, ok := proxy.firewall[ns]; ok {
+			for uid := range ref.namespaceFirewalls {
+				if uid == policy.UID {
+					proxy.firewall[ns].namespaceFirewalls[uid] = firewall
+					updated = true
+					break
+				}
+			}
+		}
+		if !updated {
+			if _, ok := proxy.firewall[ns]; !ok {
+				item := proxyFirewallItem{}
+				item.namespaceFirewalls = make(map[ktypes.UID][]firewallItem)
+				item.activePolicy = nil
+				proxy.firewall[ns] = item
+			}
+			proxy.firewall[ns].namespaceFirewalls[policy.UID] = firewall
+		}
 	} else {
-		delete(proxy.firewall, policy.Namespace)
+		if _, ok := proxy.firewall[ns]; ok {
+			delete(proxy.firewall[ns].namespaceFirewalls, policy.UID)
+		}
+	}
+
+	// Set active policy for the namespace
+	if ref, ok := proxy.firewall[ns]; ok {
+		if len(ref.namespaceFirewalls) == 1 {
+			for uid := range ref.namespaceFirewalls {
+				ref.activePolicy = &uid
+				proxy.firewall[ns] = ref
+				glog.Infof("Applied firewall egress network policy: %q to namespace: %q", uid, ns)
+			}
+		} else {
+			// We only allow one policy per namespace otherwise it's hard to determine which policy to apply first
+			glog.Errorf("Found multiple egress policies, dropping all firewall rules for namespace: %q", ns)
+			ref.activePolicy = nil
```

Comment:
Hm... this is the opposite of the behavior in controller.go, where having multiple EgressNetworkPolicies means all traffic gets dropped, rather than all traffic being allowed.